### PR TITLE
Use ES6 modules, use rollup for module bundling

### PR DESCRIPTION
**Pull request by simon04**
* modularize the code using [ES6 modules](http://exploringjs.com/es6/ch_modules.html)
* use [rollup](https://github.com/rollup/rollup) for bundling (replaces browserify)
* separate holiday file per country in `holidays/`
* run `make opening_hours.js` to get the bundled file w/o embedded dependencies

### CODE REVIEW

**Comment by ypid** [comment]
Thanks very much for the PR! I guess I should do a release before I review and merge this :wink:

**Comment by pke** [comment]
Looks interesting. I was planing a code split myself, but using webpack. Basically I think this library should be componentized in a way that the base file just uses the holidays and translations already loaded. Meaning when someone wants to include the script one has to include the required holidays and translations before the core script file. The holiday and language files will register themselves globally under opening_hours.holidays etc.

In SPA based the holidays and languages could then be imported first and opening_hours last:
```es6
import "opening_hours/holidays/de"
import oh from "opening_hours/core"

// -- or import all holidays
import "opening_hours/holidays"
import oh from "opening_hours/core"

// -- or import all at once
import oh from "opening_hours"
```

What u think?

**Comment by ypid** [comment]
I finally [enabled Travis CI testing for this project](https://github.com/opening-hours/opening_hours.js/pull/201) and [made a release](https://github.com/opening-hours/opening_hours.js/pull/202).

Can you try to get the build passing? And don’t worry about the merge conflict.

File: holidays/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export {default as at} from './at.js';
```
Comment (ypid) [inline holidays/index.js]: I would prefer to have the holidays as simple JSON files for interoperability reasons. Seems like [rollup-plugin-json](https://github.com/rollup/rollup-plugin-json) could come in handy for this. Does this make sense and can you try it?

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const banner = readFileSync('./banner.js', 'utf-8');
+const dependencies = process.env.DEPS === 'YES';
+
+export default {
+    banner: banner,
+    entry: './index',
+    format: 'cjs',
+    plugins: dependencies ? [
+        nodeResolve(),
+        common()
```
Comment (ypid) [inline rollup.config.js]: Can you end all list items with a comma please? This makes moving items around really easy and I hate that JSON does not allow this.

Comment (shouze) [inline rollup.config.js]: @ypid so can be enforced on ci by running eslint with the associated eslintrc & dev requirement packages

File: holidays/index.js (hunk 1 above)
Comment (ypid) [inline holidays/index.js]: I just found that there is also a [YAML plugin](https://github.com/rollup/rollup-plugin-yaml) :heart: So I would prefer YAML. But I can also convert that later.

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import common from 'rollup-plugin-commonjs';
+import nodeResolve from 'rollup-plugin-node-resolve';
+
+const banner = readFileSync('./banner.js', 'utf-8');
+const dependencies = process.env.DEPS === 'YES';
+
+export default {
+    banner: banner,
+    entry: './index',
+    format: 'cjs',
```
Comment (ypid) [inline rollup.config.js]: I guess you could also add: `moduleName: 'opening_hours',` so that `--format iife` works directly.

**Comment by simon04** [comment]
@ypid, thank you for the review. I hopefully addressed all comments plus performed the (cumbersome) extra tasks converting the holiday definitions to YAML.

@pke, let's address this in a separate issue/PR in order to not end up with one monster PR which is impossible to review.

**Comment by ypid** [comment]
@simon04 Thanks so much for you work, also with updating the holiday definitions! I have been working on a Python script to update YAML format over the last few days to solve #192 so this is coming together nicely.

But the older NodeJS versions, I guess https://babeljs.io/ can be used for that?

**Comment by ypid** [comment]
Could you make a changelog entry now that you rebased on latest master? :wink:

**Comment by ypid** [comment]
You can drop support for legacy versions if you want to. According to the [LTS schedule](https://github.com/nodejs/LTS#lts-schedule) they are EOL anyway. I just supported them because interestingly when I tried with the latest NodeJS release, basically nothing broke.

**Comment by simon04** [comment]
Node.js: 0.12 compatibility is back. Had to amend my commit for CHANGELOG a few times to get the syntax for links right.

**Comment by simon04** [comment]
Jay, all tests pass :smile:

**Comment by ypid** [comment]
Awesome! I will be working on getting this merged the next days.

**Comment by simon04** [comment]
I rebased again and introduced an initial renaming commit 378b226 to retain the Git history for `index.js`. Additionally, I converted the `locales/` strings to YAML format in 
	880b032.

**Comment by shouze** [comment]
@ypid let's go for a merge?

**Comment by ypid** [comment]
Sure, working on it.

**Comment by pke** [comment]
@simon04 now that this is merged, how can we make use of it in es6 to only include what one needs. Namely don't include all the holidays. 

Also maybe u have a clue why the peer dependency of momentjs is still installed underneath this modules node folder even if my app already provided a newer momentjs library.

@ypid what for is moment js used here? For date calculation or merely to display dates and times? If the latter then maybe we could split that out in a new bundle.
If moment js is used for parsing dates or times then there are smaller es6 friendly Libs available . I'll open a new issue for that.

**Comment by ypid** [comment]
> now that this is merged, how can we make use of it in es6 to only include what one needs. 

What you are asking for is already tracked here: #143 which should be used for discussion. This PR is not the right place.

> what for is moment js used here?

The source is your friend. Currently only to get the list of weekdays in the configured language. But it is optional.

**Pull request merged**